Microsoft.Toolkit.Uwp/Helpers/ObjectStorage/ApplicationDataStorageHelper.CacheFolder.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Helpers;
+using Windows.Storage;
+
+namespace Microsoft.Toolkit.Uwp.Helpers
+{
+    /// <summary>
+    /// An extension of ApplicationDataStorageHelper with additional features for interop with the LocalCacheFolder.
+    /// </summary>
+    public partial class ApplicationDataStorageHelper
+    {
+        /// <summary>
+        ///  Gets the storage folder.
+        /// </summary>
+        public StorageFolder CacheFolder => AppData.LocalCacheFolder;
+
+        /// <summary>
+        /// Determines if a directory item already exists in the LocalCacheFolder.
+        /// </summary>
+        /// <param name="itemName">Key of the file.</param>
+        /// <returns>True if an item exists.</returns>
+        public Task<bool> CacheItemExistsAsync(string itemName)
+        {
+            return ItemExistsAsync(CacheFolder, itemName);
+        }
+
+        /// <summary>
+        /// Retrieves an object from a file in the LocalCacheFolder.
+        /// </summary>
+        /// <typeparam name="T">Type of object retrieved.</typeparam>
+        /// <param name="filePath">Path to the file that contains the object.</param>
+        /// <param name="default">Default value of the object.</param>
+        /// <returns>Waiting task until completion with the object in the file.</returns>
+        public Task<T> ReadCacheFileAsync<T>(string filePath, T @default = default)
+        {
+            return ReadFileAsync<T>(CacheFolder, filePath, @default);
+        }
+
+        /// <summary>
+        /// Retrieves the listings for a folder and the item types in the LocalCacheFolder.
+        /// </summary>
+        /// <param name="folderPath">The path to the target folder.</param>
+        /// <returns>A list of file types and names in the target folder.</returns>
+        public Task<IList<Tuple<DirectoryItemType, string>>> ReadCacheFolderAsync(string folderPath)
```

Comment:
```cs
        public Task<IList<(DirectoryItemType, string)>> ReadCacheFolderAsync(string folderPath)
```
Ah, @shweaver-MSFT this is what @Sergio0694 was talking about with the `Tuple`. Not about using it, but that it's like using `Boolean` instead of `bool`. `Tuple` is the old class/way of doing it, and there's just built-in syntax for it now.

I made the exact same mistake this past weekend on my own project. 🙂

Reply:
Nice 😎 TIL

Reply:
Also note that as I mentioned in my comment above, elements in a value tuple can be named. That is, you could change this API to be like:

```csharp
Task<IEnumerable<(DirectoryItemType ItemType, string Path)>> ReadFolderAsync(string folderPath);
```

And consumers could refer to each component in the tuple by name, which is very convenient and easier to read 🙂

> "Not about using it, but that it's like using Boolean instead of bool. Tuple is the old class/way of doing it, and there's just built-in syntax for it now."

@michael-hawker just to be super nitpicky here, note that the comparison with `Boolean` isn't technically applicable in this case. `Boolean` is the underlying type in the runtime, and `bool` is just a C# alias for it. Whenever you use `bool`, you're just hardcoding `System.Boolean`, just like if you do `int` you're using `System.Int32`, but it's the same type. The difference with tuples instead is that the "old" tuple, `System.Tuple<,>` is a completely different type than the one matched by the new `(a, b)` syntax, which corresponds to `System.ValueTuple<,>`. Crucially, the latter is also a mutable value type, whereas the former is an immutable reference type. So it's not just a stylistic difference, it's also semantically different and not binary compatible either 👍